src/Execute/executeQuickInput.ts

```diff
@@ -0,0 +1,151 @@
+ * limitations under the License.
+ */
+
+import * as vscode from 'vscode';
+import {globalBackendMap} from '../Backend/Backend';
+import { Backend } from '../Backend/API';
+import { MultiStepInput } from '../Utils/MultiStepInput';
+import { exec } from 'child_process';
+import { appendFileSync } from 'fs';
+
+export async function runInferenceQuickInput(context: vscode.ExtensionContext) {
+  interface State {
+    backend: Backend;
+    modelPath: vscode.Uri;
+    inputSpec: string;
+    infer: vscode.QuickPickItem;
+  }
+
+  async function collectInputs() {
+    const state = {} as Partial<State>;
+    await MultiStepInput.run(input => pickBackend(input, state));
+    return state as State;
+  }
+
+  const title = 'Choose Executor Toolchain';
+
+  async function pickBackend(input: MultiStepInput, state: Partial<State>) {
+    const backends: vscode.QuickPickItem[] = Object.keys(globalBackendMap).map(label => ({ label }));
+    state.infer = await input.showQuickPick({
+      title,
+      step: 1,
+      totalSteps: 3,
+      placeholder: 'Pick Executor',
+      items: backends,
+      shouldResume: shouldResume
+    });
+    state.backend = globalBackendMap[state.infer.label];
+    return (input: MultiStepInput) => selectInputModel(input, state);
+  }
+
+  // TODO: Use quickPick window with fast grep child process
+  async function selectInputModel(input: MultiStepInput, state: Partial<State>) {
+    if (state.backend === undefined) {
+      throw Error('Backend is undefined');
+    }
+
+    let backendName: string = state.backend.name();
+    const executor = state.backend.executor();
+    let filter: { [name: string]: string[]} = {};
+    // List files which are filtered with executable extensions
+    if (executor) {
+      filter = {
+        backendName: executor.getExecutableExt()
+      };
+    }
+
+    const fileUri = await vscode.window.showOpenDialog({
+      title: `${title} (2/3)`,
+      canSelectMany: false,
+      openLabel: 'Select Model to Infer',
+      filters: filter
+    });
+    if(fileUri && fileUri[0]) {
+      state.modelPath = fileUri[0];
+    }
+    else {
+      console.log("No model has been selected");
+      state.modelPath = undefined;
+    }
+
+    return (input: MultiStepInput) => selectInputSpec(input, state);
+  }
+
+  // TODO: enable the backend-driven option steps by backend extension
+  async function selectInputSpec(input: MultiStepInput, state: Partial<State>) {
+    const inputSpecKeys: vscode.QuickPickItem[] = ["any", "non-zero", "positive"].map(label => ({ label }));
+
+    state.infer = await input.showQuickPick({
+      title,
+      step: 3,
+      totalSteps: 3,
+      placeholder: 'Select Random Input Spec',
+      items: inputSpecKeys,
+      shouldResume: shouldResume
+    });
+    state.inputSpec = state.infer.label;
+  }
+
+  async function runInference (state: State){
+    let cmd = state.backend.executor()?.runInference(state.modelPath.path, ["--input-spec", state.inputSpec]);
```

Comment:
if executor is undefined?

Reply:
I'll add error handler. User may notice the error by notification which you can normally find in right bottom on your vscode window.

```ts
let cmd = state.backend.executor()?.runInference(state.modelPath.path, ["--input-spec", state.inputSpec]);
// For example,
if (cmd === undefined)  { 
  console.log("Backend executor is undefined."); 
  Balloon("Your Backend Executor is undefined!");
  return
}
```